src/routes/index.js

```diff
@@ -31,33 +37,58 @@ export default function BaseRouter() {
   };
 
   return (
-    <>
-      <Switch>
-        {RouteConfig.auth.map((route, index) => {
-          return getAuthenticatedRoute(route, index);
-        })}
-        {!isLoggedIn
-          ? RouteConfig.orderPlacer.map((route, index) => {
-              return <Route key={index} component={() => route.component()} exact path={route.path} />;
-            })
-          : null}
-        {RouteConfig.common.map((route, index) => {
-          return <Route key={index} component={() => route.component()} exact path={route.path} />;
-        })}
-        {!isLoggedIn &&
-          RouteConfig.admin.map((route, index) => {
-            return <Route key={index} component={() => route.component()} exact path={route.path} />;
-          })}
-        {isLoggedIn ? (
-          <Route>
-            <Redirect to="/home" />
-          </Route>
-        ) : (
-          <Route>
-            <Redirect to="/login" />
-          </Route>
-        )}
-      </Switch>
-    </>
+    <Switch>
+      <Route component={Unauthorize} exact path="/unauthorize" />
+      {routeConfig.auth.map((route, index) => {
+        return getAuthenticatedRoute(route, index);
+      })}
+      {routeConfig.common.map((route, index) => {
+        return <Route key={index} component={() => route.component()} exact path={route.path} />;
+      })}
+      {routeConfig.customer.map((route, index) => (
+        <Route
+          key={index}
+          component={() => <ValidRoute authorizedRole={user} route={route} />}
+          exact
+          path={route.path}
+        />
+      ))}
+      {routeConfig.vendor.map((route, index) => (
+        <Route
+          key={index}
+          component={() => <ValidRoute authorizedRole={vendor} route={route} />}
+          exact
+          path={route.path}
+        />
+      ))}
+      {routeConfig.admin.map((route, index) => (
+        <Route
+          key={index}
+          component={() => <ValidRoute authorizedRole={admin} route={route} />}
+          exact
+          path={route.path}
+        />
+      ))}
+      {role === admin && (
```

Comment:
There still is this thing please remove it

Reply:
outdated code